projects/elements/src/lib/lazy-elements/lazy-elements-loader.service.ts

```diff
@@ -2,42 +2,57 @@ import { Injectable } from '@angular/core';
 
 const LOG_PREFIX = '@angular-extensions/elements';
 
+export interface ElementConfig {
+  tag: string;
+  url: string;
+}
+
 @Injectable({
   providedIn: 'root'
 })
 export class LazyElementsLoaderService {
   registry: Map<string, Promise<void>> = new Map<string, Promise<void>>();
+  configs: ElementConfig[] = [];
 
   constructor() {}
 
+  addConfig(configs) {
+    this.configs = [...this.configs, ...configs];
```

Comment:
Also whenever you lazy load new module, you will get all the configs from `@Inject(ELEMENTS_CONFIG) configs ` ( so also the configs which already were added and re-add them again, which is not correct )

I really like the feature but it has to be done in a clean way...